include/net/icalendar_parser.rst

```diff
@@ -0,0 +1,38 @@
+.. _icalendar_parser_readme:
+
+iCalendar parser
+################
+
+The iCalendar parser library can be used to parse a data stream in `iCalendar`_ format, which is a data format for representing and exchanging calendaring and scheduling information.
+The library parses the calendaring information and returns parsed calendar events.
+
+The library first detects the beginning of the calendar object by locating the delimiter ``BEGIN:VCALENDAR``.
+It then parses the following calendar content fragment by fragment.
+For each calendar component that is parsed, the library sends a parsed event (:c:type:`ical_parser_evt`) to the application.
```

Comment:
What is the object for the parsed event? I couldn't find ICAL_PARSER_EVT_COMPONENT in the code.

Reply:
ICAL_PARSER_EVT_COMPONENT is out of date. The event reported by library is now ical_parser_evt, which contains event id (ical_parser_evt_id), error code (ical_parser_error_id), and parsed component(struct ical_component ).

Reply:
OK, that's what I suggested as well, so we can probably leave this as is.